app/models/environment.rb

```diff
@@ -0,0 +1,10 @@
+class Environment < ApplicationRecord
+
+  has_many :tokens
```

Comment:
`before_destroy` callback to revoke all tokens for an environment that is being removed.
```
  private
  
  def revoke_all_tokens!
    self.tokens.all.each &:revoke!
  end
```

Corresponding model spec to test all tokens for a destroyed environment are revoked:

`expect(token).to be_revoked`